sys/riscv/include/atomic.h

```diff
@@ -354,14 +354,46 @@ static __inline void
 atomic_set_ptr(volatile uintptr_t *p, uintptr_t val)
 {
 
+#ifdef notyet
 	(void)__atomic_or_fetch(p, val, __ATOMIC_RELAXED);
+#else
+	uintptr_t temp1;
+	u_long temp2;
+
+	__asm __volatile(
+		"1:	clr.c	%1, %0\n"
+		"	cgetaddr %2, %1\n"
+		"	or	%2, %2, %3\n"
+		"	csetaddr %1, %1, %2\n"
+		"	csc.c	%2, %1, %0\n"
+		"	bnez	%2, 1b\n"
+		: "+A" (*p), "=&C" (temp1), "=&r" (temp2)
+		: "r" ((ptraddr_t)val)
+		: "memory");
+#endif
 }
 
 static __inline void
 atomic_clear_ptr(volatile uintptr_t *p, uintptr_t val)
 {
 
+#ifdef notyet
 	(void)__atomic_and_fetch(p, ~val, __ATOMIC_RELAXED);
+#else
+	uintptr_t temp1;
+	u_long temp2;
+
+	__asm __volatile(
```

Comment:
We should really do this expansion in the compiler but it's kinda awkward since built-in support for RISC-V expansion only allows expanding it to a cmpxchg sequence.